django/db/models/sql/query.py

```diff
@@ -1408,13 +1408,23 @@ def build_filtered_relation_q(self, q_object, reuse, branch_negated=False, curre
     def add_filtered_relation(self, filtered_relation, alias):
         filtered_relation.alias = alias
         lookups = dict(get_children_from_q(filtered_relation.condition))
-        for lookup in chain((filtered_relation.relation_name,), lookups):
+        relation_lookup_parts, relation_field_parts, _ = self.solve_lookup_type(filtered_relation.relation_name)
+        if relation_lookup_parts:
+            raise ValueError(
+                "FilteredRelation's relation_name doesn't support lookups "
+                "(got %r)." % filtered_relation.relation_name
+            )
+        for lookup in chain(lookups):
             lookup_parts, field_parts, _ = self.solve_lookup_type(lookup)
             shift = 2 if not lookup_parts else 1
-            if len(field_parts) > (shift + len(lookup_parts)):
+            lookup_path = field_parts[:-shift]
+            for _ in relation_field_parts:
+                if lookup_path:
+                    lookup_path = lookup_path[1:]
+            if lookup_path:
```

Comment:
The above five lines can be simplified to:
```python
    ...
    shift = 2 if not lookup_parts else 1
    if len(field_parts) > (shift + len(relation_field_parts)):
    raise ValueError(...
```
so we don't need a `lookup_path` variable and iterate over the `relation_field_parts`.

Reply:
addressed!